services/provider/server/server.go

```diff
@@ -610,358 +608,172 @@ func decodeAndValidateTicket(ticket string, pubKey *rsa.PublicKey) (*services.On
-						"csi.storage.k8s.io/snapshotter-secret-name": provisionerSecretName,
-					}),
-				},
-				&pb.ExternalResource{
-					Name: fmt.Sprintf("%s-groupsnapclass", req.StorageClaimName),
-					Kind: "VolumeGroupSnapshotClass",
-					Labels: map[string]string{
-						ramenDRStorageIDLabelKey: storageID,
-					},
-					Data: mustMarshal(map[string]string{
-						"csi.storage.k8s.io/group-snapshotter-secret-name": provisionerSecretName,
-						"clusterID": clientProfile,
-						"pool":      rns.Spec.BlockPoolName,
-					}),
-				},
-				&pb.ExternalResource{
-					Kind: "ClientProfile",
-					Name: "ceph-rbd",
-					Data: mustMarshal(&csiopv1a1.ClientProfileSpec{
-						Rbd: &csiopv1a1.RbdConfigSpec{
-							RadosNamespace: cephRes.Name,
-						},
-					})},
-			)
-
-		case "CephFilesystemSubVolumeGroup":
-			subVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
-			err := s.client.Get(ctx, types.NamespacedName{Name: cephRes.Name, Namespace: s.namespace}, subVolumeGroup)
-			if err != nil {
-				return nil, status.Errorf(codes.Internal, "failed to get %s cephFilesystemSubVolumeGroup. %v", cephRes.Name, err)
+		case storagev1.SchemeGroupVersion.WithKind("StorageClass"):
+			storageClass := resource.(*storagev1.StorageClass)
+			if req.StorageClaimName == util.GenerateNameForCephBlockPoolStorageClass(storageCluster) {
+				storageClass.Parameters["radosnamespace"] = consumerConfig.GetRbdRadosNamespaceName()
+			} else if req.StorageClaimName == util.GenerateNameForCephFilesystemStorageClass(storageCluster) {
+				storageClass.Parameters["subvolumegroup"] = consumerConfig.GetSubVolumeGroupName()
 			}
-
-			provisionerSecretName := storageClaimCephCsiSecretName("provisioner", storageRequestHash)
-			nodeSecretName := storageClaimCephCsiSecretName("node", storageRequestHash)
-			cephfsStorageClassData := map[string]string{
-				"clusterID":          getSubVolumeGroupClusterID(subVolumeGroup),
-				"subvolumegroupname": subVolumeGroup.Name,
-				"fsName":             subVolumeGroup.Spec.FilesystemName,
-				"pool":               subVolumeGroup.GetLabels()[ocsv1alpha1.CephFileSystemDataPoolLabel],
-				"csi.storage.k8s.io/provisioner-secret-name":       provisionerSecretName,
-				"csi.storage.k8s.io/node-stage-secret-name":        nodeSecretName,
-				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
+			if storageClass.Name == req.StorageClaimName {
+				extR = append(extR,
+					&pb.ExternalResource{
+						Name:        storageClass.Name,
+						Kind:        "StorageClass",
+						Labels:      storageClass.Labels,
+						Annotations: storageClass.Annotations,
+						Data:        mustMarshal(storageClass.Parameters),
+					})
 			}
 
-			storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
-			if err != nil {
-				return nil, err
+		case csiopv1a1.GroupVersion.WithKind("ClientProfile"):
+			clientProfile := resource.(*csiopv1a1.ClientProfile)
+			var clientProfileName string
```

Comment:
this func was called more than a couple of times, either store the value in a var and use it multiple times or you directly compare to `ocs-storagecluster-ceph-rbd` or `ocs-storagecluster-cephfs` since those were the only supported claims.